test/db/repo/manager_test.exs

```diff
@@ -240,4 +262,106 @@ defmodule Feeb.DB.Repo.ManagerTest do
       assert_receive :got_connection, 50
     end
   end
+
+  describe "monitoring logic - leasee's death" do
+    test "when process holding connection dies, connection is released", %{manager: manager} do
+      request_pid =
+        spawn_and_wait(fn ->
+          Manager.fetch_connection(manager, :write)
+          block_forever()
+        end)
+
+      # The Repo.Manager currently has an active write_1 connection leased to `request_pid`
+      state_before = :sys.get_state(manager)
+      assert state_before.write_1.busy?
+      assert state_before.write_1.caller_pid == request_pid
+
+      # Now we kill `request_pid`
+      Process.exit(request_pid, :kill)
+
+      # Give ample time for the `handle_info({:DOWN, ...})` callback to run
+      :timer.sleep(10)
```

Comment:
I agree, but I don't see a way to sync deterministically without having a hack that affects the test readability.

For example, I could instead do a GenServer.call to the Repo.Manager: singe a GenServer is sequential, the call will only return after `handle_info` has executed -- but it would still have the race condition of the call arriving *before* the `:DOWN` signal, so we are back to square one.

Your suggestion of trapping exist doesn't apply -- or maybe it does, but not in the way you suggested: I can wait for the `:DOWN` message in the spawned pid and, once it arrives, I send a message to the test pid so it can proceed. That would work, but IMO it's not worth the readability downsides. I'll do that only if this test proves to flake. So far, it hasn't.